gnomad/utils/constraint.py

```diff
@@ -877,3 +839,344 @@ def compute_expected_variants(
+    expected number of variants times a varying parameter ranging between 0 and 2 (the observed:expected ratio is typically between 0 and 1, so we want to extend the upper bound of the confidence interval to capture this). The
+    cumulative density function of the Poisson distribution density is computed and the
+    value of the varying parameter is extracted at points corresponding to `alpha`
+    (defaults to 5%) and 1-`alpha`(defaults to 95%) to indicate the lower and upper
+    bounds of the confidence interval.
+
+    The following annotations are in the output StructExpression:
+    - lower - the lower bound of confidence interval
+    - upper - the upper bound of confidence interval
+
+    :param obs_expr: Expression for the observed variant counts of pLoF, missense, or
+        synonymous variants in `ht`.
+    :param exp_expr: Expression for the expected variant counts of pLoF, missense, or
+        synonymous variants in `ht`.
+    :param alpha: The significance level used to compute the confidence interval.
+        Default is 0.05.
+    :return: StructExpression for the confidence interval lower and upper bounds.
+    """
+    # Set up range between 0 and 2.
+    range_expr = hl.range(0, 2000).map(lambda x: hl.float64(x) / 1000)
+    range_dpois_expr = range_expr.map(lambda x: hl.dpois(obs_expr, exp_expr * x))
+
+    # Compute cumulative density function of the Poisson distribution density.
+    cumulative_dpois_expr = hl.cumulative_sum(range_dpois_expr)
+    max_cumulative_dpois_expr = cumulative_dpois_expr[-1]
+    norm_dpois_expr = cumulative_dpois_expr.map(lambda x: x / max_cumulative_dpois_expr)
+
+    # Extract the value of the varying parameter within specified range.
+    lower_idx_expr = hl.argmax(
+        norm_dpois_expr.map(lambda x: hl.or_missing(x < alpha, x))
+    )
+    upper_idx_expr = hl.argmin(
+        norm_dpois_expr.map(lambda x: hl.or_missing(x > 1 - alpha, x))
+    )
+    return hl.struct(
+        lower=hl.if_else(obs_expr > 0, range_expr[lower_idx_expr], 0),
+        upper=range_expr[upper_idx_expr],
+    )
+
+
+def calculate_raw_z_score(
+    obs_expr: hl.expr.Int32Expression,
+    exp_expr: hl.expr.Float32Expression,
+) -> hl.expr.StructExpression:
+    """
+    Compute the signed raw z-score using observed and expected variant counts.
+
+    The raw z-scores are positive when the transcript had fewer variants than expected,
+    and are negative when transcripts had more variants than expected.
+
+    :param obs_expr: Observed variant count expression.
+    :param exp_expr: Expected variant count expression.
+    :return: StrucExpression for the raw z-score.
+    """
+    chisq_expr = (obs_expr - exp_expr) ** 2 / exp_expr
+    return hl.sqrt(chisq_expr) * hl.if_else(obs_expr > exp_expr, -1, 1)
+
+
+def add_constraint_flags(
+    exp_expr: hl.expr.Float32Expression, outlier_expr: hl.expr.BooleanExpression
+) -> hl.expr.SetExpression:
```

Comment:
I modified to only return a dict so that the no_variants flag can be added and then the filters created with from gnomad.utils.filtering import add_filters_expr